File: Array/zhangjianan/20200615队列实现栈/MyStack.java
class MyStack {
    private Queue<Integer> q1=new LinkedList<>();
    private Queue<Integer> q2=new LinkedList<>();
    private int top;
    /** Initialize your data structure here. */
    public MyStack() {

    }
    
    /** Push element x onto stack. */
    public void push(int x) {
            q1.add(x);
            top=x;
    }
    
    /** Removes the element on top of the stack and returns that element. */
    public int pop() {
            while(q1.size()>1){
                top=q1.remove();
                q2.add(top);
            }
            int popData = q1.remove();
            Queue<Integer>  temp=q1;
            q1=q2;
            q2=temp;
            return popData;
    }
    
    /** Get the top element. */
    public int top() {
        return top;
    }
    
    /** Returns whether the stack is empty. */
    public boolean empty() {
        return q1.size()<1;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: Array/zhangjianan/20200615队列实现栈/MyStack.kt
class MyStack() {
    var q1=LinkedList<Int>()
     var q2=LinkedList<Int>()
     var top:Int=0
    /** Initialize your data structure here. */


    /** Push element x onto stack. */
    fun push(x: Int) {
		q1.add(x)
		top=x
    }

    /** Removes the element on top of the stack and returns that element. */
    fun pop(): Int {
        while(q1.size>1){
           top= q1.remove()
            q2.add(top)
        }
        val popData=q1.remove()
        val temp=q1;
        q1=q2
        q2=temp
        return popData
    }

    /** Get the top element. */
    fun top(): Int {
        return top
    }

    /** Returns whether the stack is empty. */
    fun empty(): Boolean {
        return q1.size<1
    }

}

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */